### PR TITLE
[prometheus-postgre-exporter] new securityContext for POD

### PR DESCRIPTION
**Pull request by sankalp28**
Yes

#### What this PR does / why we need it
I would like to adjust the securityContext of the containers inside the deployment.yaml template, but i find no place to do so correctly.
There is already a securityContext, but that is for Pod level not container level.
Pod Security context doesn't allow the same keys to be used as the containerSecurityContext , eg capabilities, readonlyfilesystem etc.

#### Which issue this PR fixes
https://github.com/prometheus-community/helm-charts/issues/2685
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-postgres-exporter/templates/deployment.yaml [hunk 1]
```diff
@@ -31,6 +31,8 @@ spec:
 {{- end }}
     spec:
       serviceAccountName: {{ template "prometheus-postgres-exporter.serviceAccountName" . }}
+      securityContext:
```
Comment (monotek) [inline charts/prometheus-postgres-exporter/templates/deployment.yaml:34]: Please use "with", so security context is only set if actually used.

File: charts/prometheus-postgres-exporter/values.yaml [hunk 2]
```diff
@@ -94,9 +107,18 @@ networkPolicy:
   # Set labels for the NetworkPolicy
   labels: {}
 
-securityContext: {}
+securityContext:
```
Comment (monotek) [inline charts/prometheus-postgres-exporter/values.yaml:110]: Please keep default

**Review by monotek — CHANGES_REQUESTED** [review]
Please raise chart version

**Comment by zanhsieh** [comment]
@sankalp28 
Please sign DCO and bump the version in Chart.yaml.

File: charts/prometheus-postgres-exporter/templates/deployment.yaml [hunk 3]
```diff
@@ -31,6 +31,7 @@ spec:
 {{- end }}
     spec:
       serviceAccountName: {{ template "prometheus-postgres-exporter.serviceAccountName" . }}
+{{- if .Values.podSecurityContext }}     
       securityContext:
 {{ toYaml .Values.podSecurityContext | indent 8 }}
```
Comment (zanhsieh) [inline charts/prometheus-postgres-exporter/templates/deployment.yaml:36]:
```suggestion
      {{- with .Values.podSecurityContext }}     
      securityContext:
        {{- toYaml . | nindent 8 }}
      {{- end }}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Comment by zanhsieh** [comment]
@sankalp28
For DCO signing, please scroll down this PR page. In `Merging is blocked`, you should see a line `DCO Action required after xxs - DCO         Required       Details`. Click `Details` for instruction. Thanks for your cooperation and contribution.

**Comment by sankalp28** [comment]
i tried to amend the commit message for the one missing with DCO , in case it doesn't work the way t needs to, please provide steps to get the commit in DCO signed off manner.

File: charts/prometheus-postgres-exporter/values.yaml [hunk 4]
```diff
@@ -24,6 +24,17 @@ service:
   labels: {}
   annotations: {}
 
+podSecurityContext: {}
+  # runAsUser: 1001
+  # runAsGroup: 1001
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # allowPrivilegeEscalation: false
+  # capabilities:
```
Comment (monotek) [inline charts/prometheus-postgres-exporter/values.yaml:33]: please check which options actualy work for podsecuritycontect.
"capabilities" for example does not.

File: charts/prometheus-postgres-exporter/Chart.yaml [hunk 5]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 appVersion: "0.10.1"
 description: A Helm chart for prometheus postgres-exporter
 name: prometheus-postgres-exporter
-version: 3.1.5
+version: 3.1.6
```
Comment (monotek) [inline charts/prometheus-postgres-exporter/Chart.yaml:5]: new feature should raise minor version

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-postgres-exporter/Chart.yaml (hunk 5 above)
Comment (sankalp28) [reply charts/prometheus-postgres-exporter/Chart.yaml:5]: appVersion: "0.10.2"
version: 3.1.6 
is this correct ?

Comment (monotek) [reply charts/prometheus-postgres-exporter/Chart.yaml:5]: version field describes version of the chart.
appversion describes version of the app / docker image (if there is a newer docker image available you might want to update that too).

minor version change would be: 3.2.6 for the version field.
See: https://semver.org/

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by zanhsieh — APPROVED** [review]

**Comment by sankalp28** [comment]
is this now resolved, i see suggested changes already merged here ?

**Comment by sankalp28** [comment]
@monotek @gianrubio Please review and let me know for further process on this , as i would need new changes in the chart and template , and use in our project.

File: charts/prometheus-postgres-exporter/values.yaml [hunk 6]
```diff
@@ -97,7 +104,15 @@ networkPolicy:
...
-
+  # runAsUser: 1001
+  # runAsGroup: 1001
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # allowPrivilegeEscalation: false
+  # capabilities:
+   # drop: ["ALL"]
+  # seccompProfile:
+    # type: RuntimeDefault
```
Comment (monotek) [inline charts/prometheus-postgres-exporter/values.yaml:115]: please add newline

```suggestion
    # type: RuntimeDefault

```

File: charts/prometheus-postgres-exporter/Chart.yaml [hunk 7]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 appVersion: "0.10.1"
 description: A Helm chart for prometheus postgres-exporter
 name: prometheus-postgres-exporter
-version: 3.1.5
+version: 3.2.6
```
Comment (monotek) [inline charts/prometheus-postgres-exporter/Chart.yaml:5]: version jump looks odd:

```suggestion
version: 3.3.0
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Pull request merged**